functional-tests/src/main/java/com/redhat/che/selenium/core/client/RhCheTestWorkspaceServiceClient.java

```diff
@@ -87,17 +97,41 @@ public Workspace createWorkspace(
   @Override
   public void start(String workspaceId, String workspaceName, TestUser workspaceOwner)
       throws Exception {
+    if (getStatus(workspaceId).equals(WorkspaceStatus.RUNNING)) {
+      LOG.info("Workspace is running - no need to start it.");
+      return;
+    }
     try {
       this.cheStarterWrapper.startWorkspace(workspaceId, workspaceName, token);
       waitStatus(workspaceName, owner.getName(), WorkspaceStatus.RUNNING);
       LOG.info("Workspace " + workspaceName + "is running.");
     } catch (Exception e) {
-
       LOG.error("Failed to start workspace \"" + workspaceName + "\": " + e.getMessage(), e);
       throw e;
     }
   }
 
+  @Override
+  public void waitStatus(String workspaceName, String userName, WorkspaceStatus expectedStatus)
+      throws Exception {
+    WorkspaceStatus status = null;
+    for (int i = 0; i < 600; i++) {
+      status = getByName(workspaceName, userName).getStatus();
+      if (status == expectedStatus) {
+        return;
+      } else {
+        if (expectedStatus == WorkspaceStatus.RUNNING && status == WorkspaceStatus.STOPPED) {
+          break;
+        }
+        WaitUtils.sleepQuietly(1);
+      }
+    }
+
+    throw new IllegalStateException(
+        format(
+            "Workspace %s, status=%s, expected status=%s", workspaceName, status, expectedStatus));
```

Comment:
is it possible to use format `IllegalStateException("Workspace {}, status={}, expected status={}", worksapceName, status, expectedStatus);` ? is the `format()` necessary?

Reply:
I took it from parent method which I'm overriding. Have no strong feelings about that. Will change it.

Reply:
Just investigate a bit - if I use {} I still need to use String.format method, don't I? The only thing that I can do is to concatenate it directly with ```+```.

Reply:
dismissed - leave it in original format :)

Reply:
@Katka92 I believe you are correct, `IllegalStateException` does not have any constructor with varargs AFAIK